### DISCUSSION TITLE
Envoyfilter lua not allow  replace header :authority?

### DISCUSSION THREAD

**Discussion by Deraiven** [istio/istio#52817] [Q&A]
i need map a domain to local service, i use lua to rewrite the headers with lua

      `            
      local Host = request_handle:headers():get(":authority")
                    
      if string.match(Host, "backoffice.test%d+.shub.us") then
                    
        request_handle:headers():replace(":authority", "backoffice-v2-bff")
     end

`
but i get error 
how to rewrite the :authority header?

**Comment by hzxuzhonghu** [comment]
Why do you use envoyfiter to rewrite authority, VirtualServce provided a native API `HTTPRewrite`
